source/fe/fe_simplex_p.cc

```diff
@@ -757,7 +757,6 @@ FE_SimplexP<dim, spacedim>::hp_line_dof_identities(
   const FiniteElement<dim, spacedim> &fe_other) const
 {
   AssertDimension(dim, 2);
-  Assert(this->degree <= 2, ExcNotImplemented());
```

Comment:
Your PR title says "Add a bit of documentation", but you are removing this assert here. Is this intentional?

Reply:
Yes, actually. I think that when the class was written, that assertion was put into every function. But the implementation is actually correct even for cases of higher-order elements.